jupyter_client/jsonutil.py

```diff
@@ -91,9 +92,19 @@ def squash_dates(obj):
 
 
 def date_default(obj):
-    """default function for packing datetime objects in JSON."""
+    """DEPRECATED: Use jupyter_client.jsonutil.json_default"""
+    warnings.warn(
+        "date_default is deprecated. Use jupyter_client.jsonutil.json_default.", stacklevel=2
```

Comment:
Would you care to add a `since version XXX`, I don't know if you want to do 7.0 or backport. But I think that will help users to transition.

Reply:
I wouldn't care :)

Should we do backports to `6.x.x`? I would actually not mind doing it, and it's not that big of a change, and not backward-incompatible.